clusters/clusters_api.go

```diff
@@ -542,8 +542,8 @@ func (a ClustersAPI) Edit(cluster Cluster) (info ClusterInfo, err error) {
 		// it's already running or terminated, so we're safe to edit
 		break
 	case ClusterStatePending, ClusterStateResizing, ClusterStateRestarting:
-		// let's wait tiny bit, so we return RUNNING cluster info
```

Comment:
Can you have two branches: one where you wait for terminated, the other where you wait for running.

creating a method with duplicate logic is not a good idea